Cubical/Foundations/EquivCharacterization.agda

```diff
@@ -0,0 +1,61 @@
+{-
+
+More theory about equivalences/different characterizations of equivalence
+
+- 'cong f' is an equivalence, if f is an equivalence
+
+-}
+{-# OPTIONS --cubical --safe #-}
+
+module Cubical.Foundations.EquivCharacterization where
+
+open import Cubical.Core.Everything
+
+open import Cubical.Foundations.Univalence
+open import Cubical.Foundations.Equiv
+
+isEquivCong : ∀ {ℓ} {A B : Set ℓ} {x y : A} (e : A ≃ B) → isEquiv (λ (p : x ≡ y) → (cong (e .fst) p))
+isEquivCong e = EquivJ (λ (B' A' : Set _) (e' : A' ≃ B') →
+                         (x' y' : A') → isEquiv (λ (p : x' ≡ y') → cong (e' .fst) p))
+                       (λ _ x' y' → idIsEquiv (x' ≡ y')) _ _ e _ _
+
+congEquiv : ∀ {ℓ} {A B : Set ℓ} {x y : A} (e : A ≃ B) → (x ≡ y) ≃ (e .fst x ≡ e .fst y)
+congEquiv e = ((λ (p : _ ≡ _) → cong (e .fst) p) , isEquivCong e)
+
+{-
+  Everything about path split equivalences is from https://arxiv.org/abs/1706.07526
+-}
+record isPathSplitEquiv {ℓ} {A B : Set  ℓ} (f : A → B) : Set ℓ where
+  field
+    s : B → A 
+    isSection : section f s
+    isSectionCong : (x y : A) → Σ[ s' ∈ (f(x) ≡ f(y) → x ≡ y) ] section (cong f) s'
+
+idIsPathSplitEquiv : ∀ {ℓ} {A : Set ℓ} → isPathSplitEquiv (λ (x : A) → x)
+idIsPathSplitEquiv = record {
```

Comment:
This would look nicer if it was defined using copatterns: https://agda.readthedocs.io/en/v2.5.2/language/copatterns.html

This should also have better computational behavior (as Agda doesn't eagerly unfold things defined using copatterns).

For a concrete example see:  https://github.com/agda/cubical/blob/master/Cubical/Data/BinNat/BinNat.agda#L282